### PR TITLE
[Feature] Add TextSeal watermarking support

### PR DESCRIPTION
**Pull request by NihPat95**
## Motivation

Add opt-in TextSeal watermarking to SGLang text generation. The implementation embeds statistically detectable evidence during sampling while preserving the ordinary sampling path when watermarking is disabled.

## Modifications

- Added secure TextSeal configuration with key redaction and request-header support.
- Propagated watermark configuration through HTTP, OpenAI, tokenization, scheduling, and sampling layers.
- Added batched watermark state management for filtering, merging, and context updates.
- Implemented PyTorch and Triton token selectors using pinned TextSeal arithmetic.
- Added bounded `torch.topk` candidate construction to avoid sorting the full vocabulary.
- Added admission checks for unsupported execution modes.
- Added benchmark, sample-generation, and official-detector validation tools.
- Added unit, JIT, and OpenAI endpoint tests.
- Added TextSeal documentation.

The initial implementation supports native `/generate`, `/v1/chat/completions`, and `/v1/completions`, including streaming. It currently requires TP=1, nonzero temperature, and disabled CUDA graphs. Speculative decoding and disaggregated serving are rejected.

## Accuracy Tests

The official TextSeal v2 detector from pinned commit `c60d0d1da2e59f09a698438e218a07ee779b4616` was run against 50 watermarked and 50 baseline generations from `Qwen/Qwen2.5-0.5B-Instruct`.

| Metric | Result |
| --- | ---: |
| Watermarked detection rate | 50/50 (100%) |
| Baseline false-positive rate | 0/50 (0%) |
| Positive median p-value | `4.33e-11` |
| Negative median p-value | `0.5163` |
| Detection threshold | `0.01` |

Additional validation:

- OpenAI endpoint tests: **2 passed**, covering chat/completions and streaming/non-streaming requests.
- TextSeal sampler tests: **7 passed**.
- Fixed arithmetic vectors validate parity with the pinned upstream implementation.
- Batch lifecycle tests cover filtering, merging, and context updates.
- The non-watermarked path remains separate and covered by existing sampling tests.

## Speed Tests and Profiling

Test environment:

- GPU: NVIDIA Tesla T4
- Model: `Qwen/Qwen2.5-0.5B-Instruct`
- Five alternating baseline/TextSeal pairs
- 100 prompts per trial
- Concurrency: 8
- Output length: 400 tokens
- Triton attention, PyTorch sampling, TP=1, CUDA graphs disabled

| Metric | Baseline | TextSeal | Paired change |
| --- | ---: | ---: | ---: |
| Median TPOT, mean across pairs | 22.30 ms | 23.78 ms | **+6.67%** |
| Output throughput | 339.94 token/s | 318.28 token/s | **-6.37%** |

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: **Missing `run-ci` label** -- add it to run CI tests.
Latest PR Test (Extra): :x: **Blocked** -- `run-ci` is required first.
Latest PR Test (AMD ROCm 7.2): :heavy_minus_sign: **No AMD PR run found for this commit**.

### CODE REVIEW

File: benchmark/textseal/validate_detection.py
Comment (sichidambara) [inline benchmark/textseal/validate_detection.py:1]: Why do we have to do commit level checks? Cant we instead look if a detector is available?

File: docs/docs/advanced_features/textseal_watermarking.mdx
Comment (sichidambara) [inline docs/docs/advanced_features/textseal_watermarking.mdx:1]: Are we going to move away from secret loading from a file ? Does SGLang maintainers have a supported path on how we can supporting loading secrets in a safe fashion?

Comment (sichidambara) [inline docs/docs/advanced_features/textseal_watermarking.mdx:1]: Does this work for all models? do we need to add any constraints around what models this works with?

File: python/sglang/kernels/ops/sampling/textseal_selector.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+import torch
+import triton
+import triton.language as tl
+
+_BLOCK_SIZE = 8192
```
Comment (sichidambara) [inline python/sglang/kernels/ops/sampling/textseal_selector.py:7]: How are we determining the block size?
